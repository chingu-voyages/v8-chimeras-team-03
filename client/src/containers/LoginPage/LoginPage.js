import React, { Component } from "react";
import { Link } from "react-router-dom";
import "./LoginPage.scss";

class LoginPage extends Component {
  render() {
    return (
      <div className="login">
        <h1>Get tracking</h1>
        <p>Log in to your Toggl account.</p>
        <form className="form">
          <label htmlFor="email">EMAIL ADDRESS</label>
          <input type="email" id="email" />
          <label htmlFor="password">PASSWORD</label>
          <input type="password" id="password" />
          <p>Forgot password?</p>
          <button className="log-in-btn">
            LOG IN <span />
          </button>
          <div className="or">
            <div className="hr" />
            <p>OR</p>
            <div className="hr" />
          </div>
          <button className="log-in-google-btn">
            LOG IN WITH GOOGLE
            <span />
          </button>
        </form>
        <p className="no-acc">Don't have an account?</p>
        <button>
          <Link to="/signup">
            SIGN UP FOR FREE <span />
          </Link>
        </button>
      </div>
    );
  }
}

export default LoginPage;
