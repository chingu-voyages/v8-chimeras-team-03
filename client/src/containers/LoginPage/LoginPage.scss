$grey: #e3e8eb;
$green: #88cf8f;
$red: #e1393f;

.background-grey {
  background-color: $grey;
  position: absolute;
  top: 368px;
  left: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
  overflow: visible;
  @media screen and (min-width: 768px) {
    top: 353px;
  }
  @media screen and (min-width: 1280px) {
    top: 476px;
  }
}

.login {
  text-align: center;
  background-color: $green;
  overflow: auto;
  min-height: 100vh;
  h1 {
    color: white;
    font-size: 30px;
    margin-top: 117px;
    @media screen and (min-width: 768px) {
      margin-top: 124px;
    }
    @media screen and (min-width: 1280px) {
      margin-top: 169px;
      font-size: 35px;
    }
  }
  p {
    font-size: 14px;
    opacity: 0.8;
    margin-top: 20px;
    color: white;
    @media screen and (min-width: 1280px) {
      font-size: 20px;
    }
  }
  .no-acc {
    z-index: 3;
    position: relative;
    color: black;
    font-size: 14px;
    opacity: 1;
    font-weight: bold;
    margin-top: 34px;
  }
  button {
    // last button on the page
    cursor: pointer;
    z-index: 3;
    position: relative;
    width: 225px;
    height: 33px;
    border: none;
    outline: none;
    border-radius: 17px;
    background-color: white;
    margin-top: 20px;
    @media screen and (min-width: 768px) {
      width: 353px;
    }
    @media screen and (min-width: 1280px) {
      width: 518px;
      height: 49px;
      margin-top: 39px;
      margin-bottom: 39px;
    }
    a {
      text-decoration: none;
      color: black;
      font-size: 14px;
      font-weight: bold;
      @media screen and (min-width: 1280px) {
        font-size: 20px;
      }
      span {
        width: 0;
        height: 0;
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        border-left: 5px solid black;
        display: inline-block;
        margin-left: 4px;
        @media screen and (min-width: 1280px) {
          border-top: 7px solid transparent;
          border-bottom: 7px solid transparent;
          border-left: 7px solid black;
        }
      }
    }
  }
}

.form {
  z-index: 2;
  position: relative;
  display: flex;
  flex-direction: column;
  width: 316px;
  height: 402px;
  background-color: white;
  margin: 0 auto;
  padding-left: 44px;
  padding-right: 44px;
  padding-top: 34px;
  border: 1px solid #707070;
  text-align: left;
  margin-top: 50px;
  @media screen and (min-width: 768px) {
    margin-top: 80px;
    width: 441px;
    height: 447px;
  }
  @media screen and (min-width: 1280px) {
    margin-top: 82px;
    width: 653px;
    height: 706px;
    padding: 50px 65px 0 65px;
  }
  label {
    font-size: 14px;
    font-weight: bold;
    @media screen and (min-width: 1280px) {
      font-size: 22px;
    }
  }
  input {
    background-color: $grey;
    width: 228px;
    height: 36px;
    outline: none;
    border: 1px solid #707070;
    padding-left: 10px;
    font-size: 13px;
    margin-top: 9px;
    margin-bottom: 9px;
    @media screen and (min-width: 768px) {
      width: 353px;
      margin-bottom: 34px;
    }
    @media screen and (min-width: 1280px) {
      width: 522px;
      height: 54px;
      font-size: 14px;
    }
  }
  p {
    // forgot password
    font-size: 14px;
    text-decoration: underline;
    font-weight: bold;
    color: black;
    opacity: 1;
    margin-top: 34px;
    @media screen and (min-width: 768px) {
      margin-top: 16px;
    }
  }
  .log-in-btn {
    cursor: pointer;
    background-color: $red;
    border: none;
    outline: none;
    margin-top: 23px;
    width: 225px;
    height: 33px;
    border-radius: 17px;
    color: white;
    font-size: 14px;
    font-weight: bold;
    text-align: center;
    @media screen and (min-width: 768px) {
      width: 350px;
    }
    @media screen and (min-width: 1280px) {
      width: 518px;
      height: 49px;
      font-size: 20px;
      margin-top: 50px;
      margin-bottom: 29px;
    }
    span {
      width: 0;
      height: 0;
      border-top: 5px solid transparent;
      border-bottom: 5px solid transparent;
      border-left: 5px solid white;
      display: inline-block;
      margin-left: 4px;
      @media screen and (min-width: 1280px) {
        border-top: 7px solid transparent;
        border-bottom: 7px solid transparent;
        border-left: 7px solid white;
      }
    }
  }
  .log-in-google-btn {
    cursor: pointer;
    border: 1px solid black;
    outline: none;
    width: 225px;
    height: 33px;
    border-radius: 17px;
    font-size: 14px;
    font-weight: bold;
    text-align: center;
    background-color: white;
    margin-top: 18px;
    @media screen and (min-width: 768px) {
      width: 350px;
    }
    @media screen and (min-width: 1280px) {
      width: 518px;
      height: 49px;
      font-size: 20px;
      margin-top: 47px;
    }
    span {
      width: 0;
      height: 0;
      border-top: 5px solid transparent;
      border-bottom: 5px solid transparent;
      border-left: 5px solid black;
      display: inline-block;
      margin-left: 4px;
      @media screen and (min-width: 1280px) {
        border-top: 7px solid transparent;
        border-bottom: 7px solid transparent;
        border-left: 7px solid white;
      }
    }
  }
  .or {
    display: flex;
    align-items: center;
    margin: 0 auto;
    margin-top: 18px;
    opacity: 0.5;
    p {
      text-decoration: none;
      margin-top: 0;
      margin-left: 4px;
      margin-right: 4px;
      @media screen and (min-width: 1280px) {
        font-size: 20px;
      }
    }
    .hr {
      border: 1px solid black;
      width: 100px;
      height: 1px;
      @media screen and (min-width: 1280px) {
        width: 200px;
      }
    }
  }
}
