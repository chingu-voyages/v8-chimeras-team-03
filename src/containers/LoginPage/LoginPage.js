import React, { Component } from "react";
import { Link, Redirect } from "react-router-dom";
import "./LoginPage.scss";
import { auth, provider } from "../../components/Firebase/firebase";

class LoginPage extends Component {
  state = {
    redirect: false,
    email: false,
    password: false
  };

  handleLogIn = async event => {
    if (!this.canBeSubmitted) {
      event.preventDefault();
      return;
    }
    const { email, password } = event.target.elements;
    try {
      await auth.signInWithEmailAndPassword(email.value, password.value);
      this.setState({
        redirect: true
      });
    } catch (error) {
      alert(error);
    }
  };
  handleGoogleLogIn = async event => {
    event.preventDefault();
    try {
      await auth.signInWithPopup(provider);
      this.setState({
        redirect: true
      });
    } catch (error) {
      alert(error);
    }
  };
  handleEmail = event => {
    event.target.value
      ? this.setState({ email: true })
      : this.setState({ email: false });
  };
  handlePassword = event => {
    event.target.value
      ? this.setState({ password: true })
      : this.setState({ password: false });
  };
  canBeSubmitted = () => {
    const { email, password } = this.state;
    return email && password;
  };

  render() {
    const { redirect } = this.state;
    const isEnabled = this.canBeSubmitted();
    if (redirect) {
      return <Redirect to="/dashboard" />;
    }
    return (
      <div className="login">
        <h1>Get tracking</h1>
        <p>Log in to your Toggl account.</p>
        <form onSubmit={this.handleLogIn} className="form">
          <label htmlFor="email">EMAIL ADDRESS</label>
          <input type="email" id="email" onChange={this.handleEmail} />
          <label htmlFor="password">PASSWORD</label>
          <input
            type="password"
            id="password"
            onChange={this.handlePassword}
          />
          <p>Forgot password?</p>
          <button className="log-in-btn" disabled={!isEnabled}>
            LOG IN <span />
          </button>
          <div className="or">
            <div className="hr" />
            <p>OR</p>
            <div className="hr" />
          </div>
          <button
            onClick={this.handleGoogleLogIn}
            className="log-in-google-btn"
          >
            LOG IN WITH GOOGLE
            <span />
          </button>
        </form>
        <p className="no-acc">Don't have an account?</p>
        <button>
          <Link to="/signup">
            SIGN UP FOR FREE <span />
          </Link>
        </button>
      </div>
    );
  }
}

export default LoginPage;
