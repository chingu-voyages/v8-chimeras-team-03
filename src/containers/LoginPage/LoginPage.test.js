import React from "react";
import LoginPage from "./LoginPage";
import { render, shallow, mount } from "enzyme";
import { HashRouter, Link, MemoryRouter } from "react-router-dom";

it("renders <LoginPage/>", () => {
  const wrapper = render(
    <HashRouter>
      <LoginPage />
    </HashRouter>
  );
  expect(wrapper.find(".form").length).toBe(1);
});

it("checks if input if empty first time it renders", () => {
  const wrapper = shallow(
    <HashRouter>
      <LoginPage />
    </HashRouter>
  );
  expect(wrapper.find("#password").val).toEqual(undefined);
});

it("renders without crashing", () => {
  const wrapper = shallow(<LoginPage />);
  expect(wrapper).toMatchSnapshot();
});
