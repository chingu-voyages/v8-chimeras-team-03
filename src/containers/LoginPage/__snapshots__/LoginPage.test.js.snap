// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Get tracking
        </h1>,
        <p>
          Log in to your Toggl account.
        </p>,
        <form
          className="form"
        >
          <label
            htmlFor="email"
          >
            EMAIL ADDRESS
          </label>
          <input
            id="email"
            type="email"
          />
          <label
            htmlFor="password"
          >
            PASSWORD
          </label>
          <input
            id="password"
            type="password"
          />
          <p>
            Forgot password?
          </p>
          <button
            className="log-in-btn"
          >
            LOG IN 
            <span />
          </button>
          <div
            className="or"
          >
            <div
              className="hr"
            />
            <p>
              OR
            </p>
            <div
              className="hr"
            />
          </div>
          <button
            className="log-in-google-btn"
          >
            LOG IN WITH GOOGLE
            <span />
          </button>
        </form>,
        <p
          className="no-acc"
        >
          Don't have an account?
        </p>,
        <button>
          <Link
            to="/signup"
          >
            SIGN UP FOR FREE 
            <span />
          </Link>
        </button>,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Get tracking",
        },
        "ref": null,
        "rendered": "Get tracking",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Log in to your Toggl account.",
        },
        "ref": null,
        "rendered": "Log in to your Toggl account.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="email"
            >
              EMAIL ADDRESS
            </label>,
            <input
              id="email"
              type="email"
            />,
            <label
              htmlFor="password"
            >
              PASSWORD
            </label>,
            <input
              id="password"
              type="password"
            />,
            <p>
              Forgot password?
            </p>,
            <button
              className="log-in-btn"
            >
              LOG IN 
              <span />
            </button>,
            <div
              className="or"
            >
              <div
                className="hr"
              />
              <p>
                OR
              </p>
              <div
                className="hr"
              />
            </div>,
            <button
              className="log-in-google-btn"
            >
              LOG IN WITH GOOGLE
              <span />
            </button>,
          ],
          "className": "form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "EMAIL ADDRESS",
              "htmlFor": "email",
            },
            "ref": null,
            "rendered": "EMAIL ADDRESS",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "email",
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "PASSWORD",
              "htmlFor": "password",
            },
            "ref": null,
            "rendered": "PASSWORD",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "password",
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Forgot password?",
            },
            "ref": null,
            "rendered": "Forgot password?",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "LOG IN ",
                <span />,
              ],
              "className": "log-in-btn",
            },
            "ref": null,
            "rendered": Array [
              "LOG IN ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="hr"
                />,
                <p>
                  OR
                </p>,
                <div
                  className="hr"
                />,
              ],
              "className": "or",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "hr",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "OR",
                },
                "ref": null,
                "rendered": "OR",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "hr",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "LOG IN WITH GOOGLE",
                <span />,
              ],
              "className": "log-in-google-btn",
            },
            "ref": null,
            "rendered": Array [
              "LOG IN WITH GOOGLE",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Don't have an account?",
          "className": "no-acc",
        },
        "ref": null,
        "rendered": "Don't have an account?",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            to="/signup"
          >
            SIGN UP FOR FREE 
            <span />
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "SIGN UP FOR FREE ",
              <span />,
            ],
            "to": "/signup",
          },
          "ref": null,
          "rendered": Array [
            "SIGN UP FOR FREE ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Get tracking
          </h1>,
          <p>
            Log in to your Toggl account.
          </p>,
          <form
            className="form"
          >
            <label
              htmlFor="email"
            >
              EMAIL ADDRESS
            </label>
            <input
              id="email"
              type="email"
            />
            <label
              htmlFor="password"
            >
              PASSWORD
            </label>
            <input
              id="password"
              type="password"
            />
            <p>
              Forgot password?
            </p>
            <button
              className="log-in-btn"
            >
              LOG IN 
              <span />
            </button>
            <div
              className="or"
            >
              <div
                className="hr"
              />
              <p>
                OR
              </p>
              <div
                className="hr"
              />
            </div>
            <button
              className="log-in-google-btn"
            >
              LOG IN WITH GOOGLE
              <span />
            </button>
          </form>,
          <p
            className="no-acc"
          >
            Don't have an account?
          </p>,
          <button>
            <Link
              to="/signup"
            >
              SIGN UP FOR FREE 
              <span />
            </Link>
          </button>,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Get tracking",
          },
          "ref": null,
          "rendered": "Get tracking",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Log in to your Toggl account.",
          },
          "ref": null,
          "rendered": "Log in to your Toggl account.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="email"
              >
                EMAIL ADDRESS
              </label>,
              <input
                id="email"
                type="email"
              />,
              <label
                htmlFor="password"
              >
                PASSWORD
              </label>,
              <input
                id="password"
                type="password"
              />,
              <p>
                Forgot password?
              </p>,
              <button
                className="log-in-btn"
              >
                LOG IN 
                <span />
              </button>,
              <div
                className="or"
              >
                <div
                  className="hr"
                />
                <p>
                  OR
                </p>
                <div
                  className="hr"
                />
              </div>,
              <button
                className="log-in-google-btn"
              >
                LOG IN WITH GOOGLE
                <span />
              </button>,
            ],
            "className": "form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "EMAIL ADDRESS",
                "htmlFor": "email",
              },
              "ref": null,
              "rendered": "EMAIL ADDRESS",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "email",
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "PASSWORD",
                "htmlFor": "password",
              },
              "ref": null,
              "rendered": "PASSWORD",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "password",
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Forgot password?",
              },
              "ref": null,
              "rendered": "Forgot password?",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "LOG IN ",
                  <span />,
                ],
                "className": "log-in-btn",
              },
              "ref": null,
              "rendered": Array [
                "LOG IN ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="hr"
                  />,
                  <p>
                    OR
                  </p>,
                  <div
                    className="hr"
                  />,
                ],
                "className": "or",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "hr",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "OR",
                  },
                  "ref": null,
                  "rendered": "OR",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "hr",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "LOG IN WITH GOOGLE",
                  <span />,
                ],
                "className": "log-in-google-btn",
              },
              "ref": null,
              "rendered": Array [
                "LOG IN WITH GOOGLE",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
              ],
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Don't have an account?",
            "className": "no-acc",
          },
          "ref": null,
          "rendered": "Don't have an account?",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              to="/signup"
            >
              SIGN UP FOR FREE 
              <span />
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "SIGN UP FOR FREE ",
                <span />,
              ],
              "to": "/signup",
            },
            "ref": null,
            "rendered": Array [
              "SIGN UP FOR FREE ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
